tools/saqc/gen.py

```diff
@@ -0,0 +1,732 @@
+
+    for param_name, param in parameters.items():
+        if param_name in ["self", "kwargs", "store_kwargs", "ax_kwargs"]:
+            continue
+        annotation = param.annotation
+
+        if isinstance(annotation, str):
+             try:
+                 eval_context = {**globals(), **module.__dict__, **saqc.__dict__, **saqc.lib.types.__dict__, **saqc.funcs.__dict__}
+                 eval_context.update({'pd': pd, 'np': np, 'mpl': mpl, 'Union': Union, 'Literal': Literal, 'Sequence': Sequence, 'Callable': Callable, 'Any': Any, 'Tuple': Tuple, 'Dict': Dict})
+                 for m_name, m_obj in inspect.getmembers(saqc.funcs, inspect.ismodule):
+                     eval_context[m_name] = m_obj
+                 annotation = eval(annotation, eval_context)
+             except Exception:
+                 sys.stderr.write(f"Warning: Could not evaluate annotation string '{param.annotation}' for param '{param_name}' in method '{method.__name__}'. Treating as Any.\n")
+                 annotation = Any
+        elif isinstance(annotation, ForwardRef):
+             try:
+                 eval_context = {**globals(), **module.__dict__, **saqc.__dict__, **saqc.lib.types.__dict__, **saqc.funcs.__dict__}
+                 eval_context.update({'pd': pd, 'np': np, 'mpl': mpl, 'Union': Union, 'Literal': Literal, 'Sequence': Sequence, 'Callable': Callable, 'Any': Any, 'Tuple': Tuple, 'Dict': Dict})
+                 for m_name, m_obj in inspect.getmembers(saqc.funcs, inspect.ismodule):
+                     eval_context[m_name] = m_obj
+                 annotation = annotation._evaluate(eval_context, locals(), frozenset())
+             except Exception:
+                 sys.stderr.write(f"Warning: Could not evaluate ForwardRef '{param.annotation.__forward_arg__}' for param '{param_name}' in method '{method.__name__}'. Treating as Any.\n")
+                 annotation = Any
+
+        if annotation is inspect.Parameter.empty:
+             sys.stderr.write(f"Warning: Missing type annotation for parameter '{param_name}' in method '{method.__name__}'. Treating as Any.\n")
+             annotation = Any
+        
+        if isinstance(annotation, str):
+             sys.stderr.write(f"Warning: Annotation '{annotation}' for parameter '{param_name}' in method '{method.__name__}' resolved to a string. Treating as Any.\n")
+             annotation = Any
+
+        origin = get_origin(annotation)
+        args = get_args(annotation)
+        default = param.default if param.default is not inspect.Parameter.empty else None
+        label, help_text = get_label_help(param_name, param_docs)
+        param_constructor_args = {"label": label, "help": help_text}
+
+        is_python_optional = (param.default is not inspect.Parameter.empty)
+        is_union_with_none = is_union_type(annotation) and type(None) in args
+        optional = is_python_optional or is_union_with_none
+        param_constructor_args['optional'] = optional
```

Comment:
```suggestion
        param_constructor_args['optional'] = optional
        param_constructor_args['require_non_empty'] = not optional
```

only a few TextParam do not use this. There this needs to be set another way.